src/torcontrol.cpp

```diff
@@ -531,12 +531,20 @@ void TorController::auth_cb(TorControlConnection& _conn, const TorControlReply&
             SetReachable(NET_ONION, true);
         }
 
+        // Validate the target of the hidden service
+        std::string targetArg = gArgs.GetArg("-tortarget", "127.0.0.1");
```

Comment:
Looks like the port is missing in the default value ?

Reply:
If no port is specified (either on the passed in value or in this case as the default, then `GetListenPort()` is used below on L537)